src/main/java/com/jwplayer/jwplatform/rest/HttpCalls.java

```diff
@@ -0,0 +1,158 @@
+package com.jwplayer.jwplatform.rest;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.jwplayer.jwplatform.exception.JWPlatformException;
+import com.jwplayer.jwplatform.exception.JWPlatformUnknownException;
+import com.jwplayer.jwplatform.exception.MediaAPIExceptionFactory;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+public class HttpCalls {
+
+	public static JSONObject request(final String path, final Map<String, String> params, final boolean isBodyParams,
```

Comment:
Maybe we can come up with something nicer than isBodyParams. Can't believe I went with that...